lib/galaxy/tools/parameters/basic.py

```diff
@@ -676,36 +676,19 @@ def get_initial_value( self, trans, context, history=None ):
 ## it for now to avoid breaking any tools.
 
 
-class BaseURLToolParameter( ToolParameter ):
+class BaseURLToolParameter( HiddenToolParameter ):
     """
     Returns a parameter the contains its value prepended by the
     current server base url. Used in all redirects.
     """
-
-    def __init__( self, tool, input_source ):
-        input_source = ensure_input_source( input_source )
-        ToolParameter.__init__( self, tool, input_source )
-        self.value = input_source.get( 'value', '' )
-
-    def get_value( self, trans ):
-        # url = trans.request.base + self.value
-        url = url_for( self.value, qualified=True )
-        return url
-
-    def get_html_field( self, trans=None, value=None, other_values={} ):
-        return form_builder.HiddenField( self.name, self.get_value( trans ) )
-
     def get_initial_value( self, trans, context, history=None ):
-        return self.value
+        self.url_for = url_for( '', qualified=True )
```

Comment:
Can you describe that in greater detail. The client specifies the host as a header I think - you can override it with curl for instance and then ``url_for`` on the server will be modified. You can try it out by logging ``url_for("/", qualified=True)`` in controller and issuing a request like this:

```
curl -H 'Host: hacked' http://127.0.0.1:8080/
```

There may well be an attack that is prevented by having the web browser inject this instead of using a query parameter - but it is not immediately obvious to me why this would be the case - can you explain that attack?